src/gui/dialogs/preferences_dialog.cpp

```diff
@@ -1213,9 +1212,11 @@ void preferences_dialog::handle_theme_select()
 
 void preferences_dialog::handle_gui2_theme_select()
 {
-	find_widget<label>(get_window(), "restart_msg", false).set_visible(widget::visibility::visible);
 	menu_button& gui2_theme_list = find_widget<menu_button>(this, "choose_gui2_theme", false);
 	prefs::get().set_gui_theme(gui2_themes_.at(gui2_theme_list.get_value()));
+
+	gui2::init();
+	set_retval(gui2::dialogs::title_screen::RELOAD_GAME_DATA);
```

Comment:
This looks like it happens whether you select a new or the current theme. Would be good to not reload if someone just selects the current theme.

Reply:
Check added.